publish.ps1

```diff
@@ -0,0 +1,46 @@
+# **************************************************************************
+#  Copyright (c) Cloud Native Foundation.
+#  SPDX-License-Identifier: Apache-2.0
+# **************************************************************************
+
+<#
+    .SYNOPSIS
+    Publish the CloudEvents.Sdk module to PSGallery
+
+    .PARAMETER NuGetApiKey
+    PowerShell Gallery API Key to be used to publish the module
+#>
+
+param(
+    [Parameter(Mandatory = $true)]
+    [string]
+    $NuGetApiKey
+)
+
+$moduleName = 'CloudEvents.Sdk'
+$OutputDir = Join-Path $PSScriptRoot $moduleName
+
+if (-not (Test-Path $OutputDir)) {
+    New-Item -ItemType Directory -Path $OutputDir | Out-Null
+}
+
+# Build and Test before publish
+$buildScript = Join-Path $PSScriptRoot 'build.ps1'
+$failedTests = & $buildScript -OutputDir $OutputDir
+if ($failedTests -eq 0) {
+
+    # Build is successful and all tests pass
+    $env:PSModulePath += [IO.Path]::PathSeparator + $OutputDir
+
+    $localModule = Get-Module $moduleName -ListAvailable
+    $psGalleryModule = Find-Module -Name $moduleName -Repository PSGallery
+
+    # Publish only if local module version is higher than the latest available on PSGallery
```

Comment:
Where is version defined? Can we reuse the tag which triggered this workflow as the version so we avoid version drift/typos?

Reply:
the version is in the module file [psd1](https://github.com/cloudevents/sdk-powershell/blob/d4f8ff90ed48fd9258a7fba7b663c1150f3c010b/src/CloudEventsPowerShell/CloudEvents.Sdk.psd1#L12). If we update it with the tag this will mean we'll have to generate new commit which will be after the tag. Is there a solution for this?

Reply:
Ah, ok. So the flow would be:

1) Bump version in psd via PR
1) Review and merge that PR
1) Cut a release by creating a tag on that commit (or newer commits < next_tag)

Is this the only place where we have to change a version? Let's also document the individual steps to create a release, e.g. https://github.com/vmware/govmomi/blob/master/RELEASE.md